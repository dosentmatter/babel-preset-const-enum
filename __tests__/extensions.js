import { transformAsync } from '@babel/core';
import preset from '../src';

const options = {
  presets: [preset],
};

const input = `const enum Direction { Left, Right, Down, Up };
`;

it('Transforms *.ts', async () => {
  const { code: output } = await transformAsync(input, {
    ...options,
    filename: 'file.ts',
  });
  expect(output).toMatchSnapshot();
});

it('Transforms *.tsx', async () => {
  const { code: output } = await transformAsync(input, {
    ...options,
    filename: 'file.tsx',
  });
  expect(output).toMatchSnapshot();
});

it('Transforms *.*', async () => {
  const { code: output } = await transformAsync(input, {
    filename: 'file.js',
    presets: [[preset, { allExtensions: true }]],
  });
  expect(output).toMatchSnapshot();
});
