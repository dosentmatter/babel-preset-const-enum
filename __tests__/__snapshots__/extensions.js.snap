// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transforms *.* 1`] = `
"\\"use strict\\";

enum Direction {
  Left,
  Right,
  Down,
  Up,
}
;"
`;

exports[`Transforms *.ts 1`] = `
"\\"use strict\\";

enum Direction {
  Left,
  Right,
  Down,
  Up,
}
;"
`;

exports[`Transforms *.tsx 1`] = `
"\\"use strict\\";

enum Direction {
  Left,
  Right,
  Down,
  Up,
}
;"
`;
